lib/features/outline/OutlineProvider.js

```diff
@@ -0,0 +1,141 @@
+import { assign } from 'min-dash';
+
+import {
+  attr as svgAttr,
+  create as svgCreate
+} from 'tiny-svg';
+
+import {
+  is,
+  isAny
+} from '../../util/ModelUtil';
+
+import { isLabel } from '../../util/LabelUtil';
+
+import {
+  DATA_OBJECT_REFERENCE_OUTLINE_PATH,
+  DATA_STORE_REFERENCE_OUTLINE_PATH,
+  createPath
+} from './OutlineUtil';
+
+/**
+ * BPMN-specific outline provider.
+ *
+ * @implements {BaseOutlineProvider}
+ *
+ * @param {Outline} outline
+ * @param {Styles} styles
+ */
+export default function OutlineProvider(outline, styles) {
+
+  this._styles = styles;
+  outline.registerProvider(this);
+}
+
+OutlineProvider.$inject = [
+  'outline',
+  'styles'
+];
+
+/**
+ * Returns outline for a given element.
+ *
+ * @param {Element} element
+ *
+ * @return {Outline}
+ */
+OutlineProvider.prototype.getOutline = function(element) {
+
+  const OUTLINE_STYLE = this._styles.cls('djs-outline', [ 'no-fill' ]);
+
+  var outline;
+
+  if (isLabel(element)) {
+    return;
+  }
+
+  if (is(element, 'bpmn:Gateway')) {
+    outline = svgCreate('rect');
+
+    assign(outline.style, {
+      'transform-box': 'fill-box',
+      'transform': 'rotate(45deg)',
+      'transform-origin': 'center'
+    });
+
+    svgAttr(outline, assign({
+      x: 2,
+      y: 2,
+      rx: 4,
+      width: element.width - 4,
+      height: element.height - 4,
+    }, OUTLINE_STYLE));
+
+  } else if (isAny(element, [ 'bpmn:Task', 'bpmn:SubProcess', 'bpmn:Group' ])) {
+    outline = svgCreate('rect');
+
+    svgAttr(outline, assign({
+      x: -5,
+      y: -5,
+      rx: 14,
+      width: element.width + 5 * 2,
```

Comment:
I see `5 * 2` a couple of times in this PR, but it's unclear whether all of the occurrences are connected with each other. How about we extract such numbers to verbose constants so that we avoid magic numbers?

Reply:
Let's do `const MARGIN = 5`.